snuba/cli/optimize.py

```diff
@@ -49,25 +50,17 @@ def optimize(
     # will ensure that optimize is performed on all of the individual nodes for
     # that cluster.
     if clickhouse_host and clickhouse_port:
-        clickhouse_connections = [
-            ClickhousePool(
-                clickhouse_host,
-                clickhouse_port,
-                send_receive_timeout=ClickhouseClientSettings.OPTIMIZE.value.timeout,
-            )
-        ]
-    elif not local_dataset_mode():
+        connection = ClickhousePool(
+            clickhouse_host,
+            clickhouse_port,
+            send_receive_timeout=ClickhouseClientSettings.OPTIMIZE.value.timeout,
+        )
+    elif not writable_storage.get_cluster().is_single_node():
         raise click.ClickException("Provide Clickhouse host and port for optimize")
     else:
-        # In local mode, we run optimize on each cluster relevant to the provided
-        # dataset using the cluster's host/port configuration.
-        clickhouse_connections = list(
-            set(
-                storage.get_cluster().get_connection(ClickhouseClientSettings.OPTIMIZE)
-                for storage in dataset.get_all_storages()
-            )
+        connection = writable_storage.get_cluster().get_connection(
+            ClickhouseClientSettings.OPTIMIZE
         )
 
-    for connection in clickhouse_connections:
-        num_dropped = run_optimize(connection, database, table, before=today)
-        logger.info("Optimized %s partitions on %s" % (num_dropped, clickhouse_host))
+    num_dropped = run_optimize(connection, database, table, before=today)
```

Comment:
Why did we stop running optimize on all the connections we can get from the dataset?
The dataset may have multiple readable storages on different storage sets, thus with different connections.

Reply:
Hmm, isn't the `table` here always the one that is found in the writable storage of the dataset and won't exist in any of the other storages? I think i might've been responsible for adding this but looking at it again I feel like checking all the other storages is unnecessary.

Reply:
" isn't the table here always the one that is found in the writable storage"
it is, and I think it is not right. The optimize takes a dataset, thus we should be optimizing all the tables in that dataset, not only the one we write on. I think it may be working for us by coincidence because all the storages that are not writable for a dataset are writable fort another dataset. The only exception is the materialized view on which I suspect we did not run otpimize.

Reply:
Ok. I'll revert this change and attempt to fix this separately.

Reply:
On second thoughts I think this should stay. Changing the optimize function to work on all the storages is a separate and possibly bigger task given it has never worked in that way. There's little value in this partial implementation that doesn't do anything, and will need to be removed in a proper fix anyway.

Reply:
ok, mind filing a task to fix this ?

Reply:
Will do